internals/cli/cli.go

```diff
@@ -139,34 +146,18 @@ func fixupArg(optName string) string {
 	return optName
 }
 
-type clientSetter interface {
-	setClient(*client.Client)
-}
-
-type clientMixin struct {
-	client *client.Client
-}
-
-func (ch *clientMixin) setClient(cli *client.Client) {
-	ch.client = cli
-}
-
-type parserSetter interface {
-	setParser(*flags.Parser)
-}
-
 type defaultOptions struct {
 	Version func() `long:"version" hidden:"yes" description:"Print the version and exit"`
 }
 
 // Parser creates and populates a fresh parser.
 // Since commands have local state a fresh parser is required to isolate tests
 // from each other.
-func Parser(cli *client.Client) *flags.Parser {
+func Parser(c *client.Client) *flags.Parser {
```

Comment:
That looks like an unnecessary change. It's removing clarity, and creating conflicts across merges because the meaning of `c` has been shifted in the code below, and in the conversations that we have around this code, as we've experienced live yesterday.